### ISSUE TITLE
stuck when FFmpegFrameGrabber.setVideoFrameNumber

### ISSUE THREAD

**Issue by redfive-cn** [bytedeco/javacv#1297]
[http://up.v.sharedaka.com/video_hd/ochvq0GUe4jk5o5CQ48oUFVXOWnU1568253356106_1.mp4](url)
when I use  `javacv-platform 1.4.3` to grab frames from this video , it will block at `ff.setVideoFrameNumber(j * frameSpace + startFrame);` and the main thread will wait for the response forever. By the way ,there is no response when I call `FFmpegFrameGrabber.close()` or `FFmpegFrameGrabber.stop()`. How can I stop and clear the Object `FFmpegFrameGrabber` gracefully when I meet this kind of video again, do you have any suggestions ?

> Here is the code:
> import com.google.common.collect.Lists;
> import com.google.common.collect.Maps;
> import org.bytedeco.javacv.FFmpegFrameGrabber;
> import org.bytedeco.javacv.Frame;
> import org.bytedeco.javacv.FrameGrabber;
> 
> import java.awt.image.BufferedImage;
> import java.math.BigDecimal;
> import java.util.List;
> import java.util.Map;
> 
> public class Test {
> 
>     public static Map<String,Object> makeGifFromVideo(String filePath, String targetPath) throws FrameGrabber.Exception {
>         FFmpegFrameGrabber ff = FFmpegFrameGrabber.createDefault(filePath);
>         ff.start();
>         Map<String,Object> map = Maps.newHashMap();
>         String rotate = ff.getVideoMetadata("rotate");
>         double ffLength = getDivideValue(ff.getLengthInTime(), 1000000);
>         map.put("duration", (int) ffLength);
>         int j = 0;
> 
>         List<BufferedImage> picImg = Lists.newArrayList();
> 
>         int frameNumber = ff.getLengthInFrames();
>         int frameSpace = 1;
>         int frameDelay = 90;
>         int startFrame = 3;
> 
>         while (j < frameNumber) {
>             ff.setVideoFrameNumber(j * frameSpace + startFrame);
> 
>             Frame f = ff.grabImage();
> 
>             Integer rotateAngle = rotate == null ? null : Integer.valueOf(rotate);
> 
>             j++;
>         }
> 
> 
>         ff.stop();
> 
>         return map;
>     }
> 
>     private static double getDivideValue(long num1, long num2) {
>         BigDecimal bigDecimal1 = new BigDecimal(num1);
>         BigDecimal bigDecimal2 = new BigDecimal(num2);
>         BigDecimal result = bigDecimal1.divide(bigDecimal2, 2, BigDecimal.ROUND_HALF_UP);
>         return result.doubleValue();
>     }
> 
>     public static void main(String[] args) throws FrameGrabber.Exception {
>         String fileName = "ochvq0GUe4jk5o5CQ48oUFVXOWnU1568253356106_1.mp4";
>         String sourcePath = "/Users/hello/Desktop/" + fileName;
>         String destPath = "/Users/hello/dest.gif";
>         makeGifFromVideo(sourcePath, destPath);
>     }

**Comment by saudet** [comment]
Could you check what FFmpeg outputs to the console and show it to me?

**Comment by redfive-cn** [comment]
> Could you check what FFmpeg outputs to the console and show it to me?

Connected to the target VM, address: '127.0.0.1:57757', transport: 'socket'
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/Users/zhanglei/Desktop/ochvq0GUe4jk5o5CQ48oUFVXOWnU1568253356106_1.mp4':
  Metadata:
    major_brand     : iso5
    minor_version   : 512
    compatible_brands: iso6mp41
    encoder         : Lavf57.25.100
    comment         : GIFSHOW [130991480][Xiaomi|MI CC 9][9|28][pyxis][6.8.0.10648][EditorVer:fullScreen3][-1:0.0][pipeline]
  Duration: 00:00:06.28, start: 0.000000, bitrate: 5658 kb/s
    Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, smpte170m/bt470bg/smpte170m), 720x1280, 5016 kb/s, 25 fps, 25 tbr, 250k tbn, 50 tbc (default)
    Metadata:
      handler_name    : VideoHandler
    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 192 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
Disconnected from the target VM, address: '127.0.0.1:57757', transport: 'socket'

Process finished with exit code 130 (interrupted by signal 2: SIGINT)

**Comment by saudet** [comment]
Could you enable debug messages as well with `av_log_set_level(AV_LOG_DEBUG)`?

**Comment by redfive-cn** [comment]
> AV_LOG_DEBUG

like this?
`ff.setOption("av_log_set_level","AV_LOG_DEBUG");`

**Comment by saudet** [comment]
No, this: http://bytedeco.org/javacpp-presets/ffmpeg/apidocs/org/bytedeco/ffmpeg/global/avutil.html#av_log_set_level-int-

**Comment by redfive-cn** [comment]
add `avutil.av_log_set_level(AV_LOG_DEBUG);`
and here is the log:
> [NULL @ 0x7fc393bcce00] Opening '/Users/zhanglei/hello/ochvq0GUe4jk5o5CQ48oUFVXOWnU1568253356106_1.mp4' for reading
> [file @ 0x7fc39315c700] Setting default whitelist 'file,crypto'
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fc393bcce00] Format mov,mp4,m4a,3gp,3g2,mj2 probed with size=2048 and score=100
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fc393bcce00] ISO: File Type Major Brand: iso5
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fc393bcce00] Unknown dref type 0x206c7275 size 12
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fc393bcce00] Unknown dref type 0x206c7275 size 12
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fc393bcce00] found sidx time -9223372036854775808, using it for pts
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fc393bcce00] found sidx time -9223372036854775808, using it for pts
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fc393bcce00] Before avformat_find_stream_info() pos: 4092259 bytes read:98304 seeks:2 nb_streams:2
> [h264 @ 0x7fc393c67e00] nal_unit_type: 7, nal_ref_idc: 3
> [h264 @ 0x7fc393c67e00] nal_unit_type: 8, nal_ref_idc: 3
> [h264 @ 0x7fc393c67e00] nal_unit_type: 6, nal_ref_idc: 0
> [h264 @ 0x7fc393c67e00] nal_unit_type: 5, nal_ref_idc: 3
> [h264 @ 0x7fc393c67e00] Format yuv420p chosen by get_format().
> [h264 @ 0x7fc393c67e00] Reinit context to 720x1280, pix_fmt: yuv420p
> [h264 @ 0x7fc393c67e00] no picture 
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fc393bcce00] All info found
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x7fc393bcce00] After avformat_find_stream_info() pos: 3942193 bytes read:944364 seeks:4 frames:29
> Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/Users/zhanglei/Desktop/ochvq0GUe4jk5o5CQ48oUFVXOWnU1568253356106_1.mp4':
>   Metadata:
>     major_brand     : iso5
>     minor_version   : 512
>     compatible_brands: iso6mp41
>     encoder         : Lavf57.25.100
>     comment         : GIFSHOW [130991480][Xiaomi|MI CC 9][9|28][pyxis][6.8.0.10648][EditorVer:fullScreen3][-1:0.0][pipeline]
>   Duration: 00:00:06.28, start: 0.000000, bitrate: 5658 kb/s
>     Stream #0:0(und), 28, 1/250000: Video: h264 (High), 1 reference frame (avc1 / 0x31637661), yuv420p(tv, smpte170m/bt470bg/smpte170m, left), 720x1280, 0/1, 5016 kb/s, 25 fps, 25 tbr, 250k tbn, 50 tbc (default)
>     Metadata:
>       handler_name    : VideoHandler
>     Stream #0:1(und), 1, 1/44100: Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 192 kb/s (default)
>     Metadata:
>       handler_name    : SoundHandler
> [h264 @ 0x7fc392b86600] nal_unit_type: 7, nal_ref_idc: 3
> [h264 @ 0x7fc392b86600] nal_unit_type: 8, nal_ref_idc: 3
